apps/mediator/src/push-notifications/fcm/events/PushNotificationEvent.ts

```diff
@@ -1,33 +1,72 @@
+import { AgentContext } from '@credo-ts/core'
 import config from '../../../config'
 import { Logger } from '../../../logger'
-import { firebase } from '../firebase'
-
-export const sendFcmPushNotification = async (deviceToken: string, logger: Logger) => {
-  if (firebase === undefined) {
-    logger.warn('Firebase is not initialized. Push notifications are disabled.')
-    return
-  }
+import { firebaseApps } from '../firebase'
+import { PushNotificationsFcmRecord, PushNotificationsFcmRepository } from '../repository'
 
+export const sendFcmPushNotification = async (
+  agentContext: AgentContext,
+  repository: PushNotificationsFcmRepository,
+  pushNotificationRecord: PushNotificationsFcmRecord,
+  logger: Logger
+) => {
   const title = config.get('agent:pushNotificationTitle')
   const body = config.get('agent:pushNotificationBody')
 
   if (!title || !body) {
-    throw new Error('Push notification title or body is missing')
+    logger.warn('Push notification title or body is missing in configuration')
+    return
   }
 
-  try {
-    const response = await firebase.messaging().send({
-      token: deviceToken,
-      notification: {
-        title,
-        body,
-      },
-    })
+  if (pushNotificationRecord.deviceToken === null) {
+    logger.warn('Device token is null, cannot send push notification')
+    return
+  }
+
+  // Try to send using the specified projectId first
+  const firebase = firebaseApps.get(pushNotificationRecord.projectId || '')
```

Comment:
I'd maybe write this as a filter on firebase apps, that way we can basically merge the two codeblocks, as you'd always loop over fire base apps, but if there's a project id, it's just one project. This removes some duplication.